llvm/lib/Target/AIE/AIEBaseHardwareLoops.cpp

```diff
@@ -297,9 +297,12 @@ void AIEBaseHardwareLoops::expandLoopStart(LowOverheadLoop &LoLoop) {
   MachineInstr *Start = LoLoop.Start;
   MachineBasicBlock *MBB = Start->getParent();
   LLVM_DEBUG(dbgs() << "AIE Loops: ZOL loop. Expanding LoopStart.\n");
-  BuildMI(*MBB, Start, Start->getDebugLoc(), TII->get(AIE2::MOV_mv_scl),
-          AIE2::LC)
-      .addReg(Start->getOperand(0).getReg());
+
+  // We use ADD_NC, which allows PostPipeliner to tweak it by modifying the
+  // immediate value.
+  BuildMI(*MBB, Start, Start->getDebugLoc(), TII->get(AIE2::ADD_NC), AIE2::LC)
```

Comment:
We merged e33bbe18b8ed6b608cf77a9c5677d9819ae58563 as a bug fix for the trip count adjustment. As this change also overcomes the same problem, we can replace later (revert the first fix, for example).